sumologic/resource_sumologic_cse_insights_configuration_test.go

```diff
@@ -0,0 +1,101 @@
+package sumologic
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccSumologicSCEInsightsConfiguration_create(t *testing.T) {
+	var insightConfiguration CSEInsightsConfiguration
+	nLookbackDays := 10.0
+	nThreshold := 13.0
+	resourceName := "sumologic_cse_insights_configuration.insights_configuration"
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCSEInsightsConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testCreateCSEInsightsConfigurationConfig(nLookbackDays, nThreshold),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckCSEInsightsConfigurationExists(resourceName, &insightConfiguration),
+					testCheckInsightsConfigurationValues(&insightConfiguration, nLookbackDays, nThreshold),
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCSEInsightsConfigurationDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*Client)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "sumologic_cse_insights_configuration" {
+			continue
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("CSE Insights Status destruction check: CSE Insights Status ID is not set")
+		}
+
+		s, err := client.GetCSEInsightsConfiguration()
+		if err != nil {
+			return fmt.Errorf("Encountered an error: " + err.Error())
+		}
+		if s != nil {
+			if s.Threshold != 0 && s.LookbackDays != 0 {
```

Comment:
Once you do delete it, `Treshold`/`LookbackDays` should be nil, not 0, right?

Reply:
done under
https://github.com/SumoLogic/terraform-provider-sumologic/pull/274/commits/22ae5d0e51f44dc94e2ed6cb28653422ce2ef257